extensions/sql-migration/src/dialog/help/cancelFeedbackDialog.ts

```diff
@@ -0,0 +1,240 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as styles from '../../constants/styles';
+import { EventEmitter } from 'stream';
+import * as constants from '../../constants/strings';
+
+export class CancelFeedbackDialog {
+	private dialog!: azdata.window.Dialog;
+
+	private _disposables: vscode.Disposable[] = [];
+
+	private _creationEvent: EventEmitter = new EventEmitter;
+	private _isOpen: boolean = false;
+
+	private _view!: azdata.ModelView;
+
+	private _cancelReasonsList: string[] = [];
+	private _cancellationReasonSelected: string = constants.CANCEL_FEEDBACK_NO_REASON_SELECTED;
+	private _othersReasonInputBoxContainer!: azdata.FlexContainer;
+
+	private _callback!: (isCancelled: boolean, cancellationReason: string) => Promise<void>;
+
+	constructor() {
+	}
+
+	private async initializeDialog(dialog: azdata.window.Dialog): Promise<void> {
+		return new Promise<void>((resolve, reject) => {
+			dialog.registerContent(async (view) => {
+				try {
+					this._view = view;
+					const flex = this.createContainer(view);
+
+					await view.initializeModel(flex);
+
+					this._disposables.push(
+						dialog.okButton.onClick(
+							async (e) => {
+								await this._callback(true, this._cancellationReasonSelected);
+							}));
+
+					this._disposables.push(
+						view.onClosed(e =>
+							this._disposables.forEach(
+								d => { try { d.dispose(); } catch { } })));
+
+					this._creationEvent.once('done', async () => {
+						azdata.window.closeDialog(this.dialog);
+						resolve();
+					});
+				}
+				catch (ex) {
+					reject(ex);
+				}
+			});
+
+			dialog.okButton.label = "Yes";
+			dialog.cancelButton.label = "No";
```

Comment:
please create/add "yes and no" as resource strings

Reply:
Done